src/main/java/org/simplify4u/plugins/ArtifactResolver.java

```diff
@@ -112,31 +114,46 @@ private ArtifactRepositoryPolicy policyIgnoreCheckSum(ArtifactRepositoryPolicy p
      *         the artifact filter
      * @param verifyPomFiles
      *         indicator whether to verify POM file signatures as well
+     * @param verifyPlugins
+     *         indicator whether to verify build plug-ins
+     * @param verifyAtypical
+     *         indicator whether to verify "special" artifacts in non-typical locations:
+     *         annotation processors in maven-compiler-plugin.
      *
      * @return Returns set of all artifacts whose signature needs to be verified.
      */
     Set<Artifact> resolveProjectArtifacts(MavenProject project, SkipFilter filter, boolean verifyPomFiles,
-                                          boolean verifyPlugins) throws MojoExecutionException {
+                                          boolean verifyPlugins, boolean verifyAtypical) throws MojoExecutionException {
         final LinkedHashSet<Artifact> allArtifacts = new LinkedHashSet<>(
                 resolveArtifacts(project.getArtifacts(), filter, verifyPomFiles));
         if (verifyPlugins) {
+            // FIXME: applying dependency filters to build plug-ins may result in all build plug-ins being dropped because these are listed as 'runtime'-scoped.
```

Comment:
@slawekjaranowski can you double-check if you really wanted the `filter` parameter to be changed in your earlier commit?

I think that this has unintended side-effects. Most significantly: if scope is reduced to `compile` or such, no plugins will be verified due to plugins having `runtime` scope by default.

Reply:
my intention was to filter snapshot versions, maybe only snapshot should be filtered

Reply:
Thanks for confirming. I'll fix it in the PR. I'll provide a separate skipfilter for plugins which contains just the snapshot-filter or nothing.